### ISSUE TITLE
block and include aren't working correctly in diet templates

### ISSUE THREAD

**Issue by csmith1991** [rejectedsoftware/vibe.d#1344]
The following diet templates won't compile:

#template.dt

    !!! 5
    html.no-js
        body
            header
                include nav
            section
                block content

#index.dt

    extends layout

        block content
            article
                h1 Hello, World!

#nav.dt

    nav(role='navigation')

This generates the following error:

     Error: "template layout.dt line 4: The next line is indented by more than one level deeper. Please unindent accordingly.(..\\..\\AppData\\Roaming\\dub\\packages\\vibe-d-0.7.26\\source\\vibe\\templ\\diet.d:581)"

If I remove the `include nav` I getthe same exception on line 5.

Jade supports this, see the following:
* blocks: http://jade-lang.com/reference/inheritance/
* includes: http://jade-lang.com/reference/includes/

**Comment by s-ludwig** [comment]
It seems like the line number in the error message is off by one. The culprit are the three lines after the `extends layout` in index.dt. The `block` must be a sibling of `extends`, rather than a sub tree. It would probably make sense to add a specialized error message for this case.

**Comment by s-ludwig** [comment]
OT: Missed your question on IRC twice, because it was always half past 3 am here. 6 hours earlier or later would be perfect, though. Also feel free to ask even minor questions on the [forum](http://forum.rejectedsoftware.com/groups/rejectedsoftware.vibed/).

**Comment by csmith1991** [comment]
@s-ludwig: You're absolutely correct on the issue. Not sure how I missed that. :+1:

OT: I was reaching out there for a separate issue, but was able to get it answered on #d. I'll just use the forums next time. :)
